Newsletters: add 312 (2024-07-19)

harding opened the pull request:
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Clients/services @bitschmidty 
- [x] Merged PRs @Gustavojfe

bitschmidty reviewed (commented):
News section lgtm so far

LarryRuane commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,187 @@
+  threshold public key as correct, each participant verifies every other
+  participant's signed session transcript.
+
+  The goal was to design a fully generalized protocol that will be
```
```suggestion
  The goal is to design a fully generalized protocol that is
```
I'm suggesting "goal is" because "goal was" made me wonder if the goal changed and is now something different. (But this is a small point so of course feel free to ignore, as always.)

LarryRuane commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,187 @@
+  networks] support.
+
+- **BlueWallet adds silent payment send support:**
+  In the [v6.6.7][bluewallet v6.6.7] release, BlueWallet added ability to send to
```
```suggestion
  In the [v6.6.7][bluewallet v6.6.7] release, BlueWallet added the ability to send to
```

LarryRuane commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,187 @@
+  across different LN implementations.
+
+- **Moosig testing repository announced:**
+  Ledger has published a python-based testing [repository][moosig github] for using
```
```suggestion
  Ledger has published a Python-based testing [repository][moosig github] for using
```

LarryRuane commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,187 @@
+
+- **Real-time Stratum visualization tool released:**
+  The [stratum.work website][stratum.work], based on [previous research][b10c
+  nostr] displays real time Stratum messages from a variety of Bitcoin mining
```
```suggestion
  nostr] displays real-time Stratum messages from a variety of Bitcoin mining
```

LarryRuane commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,187 @@
+- **Real-time Stratum visualization tool released:**
+  The [stratum.work website][stratum.work], based on [previous research][b10c
+  nostr] displays real time Stratum messages from a variety of Bitcoin mining
+  pools, with [source available][stratum work github].
```
```suggestion
  pools, with [source code available][stratum work github].
```

vostrnad commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,187 @@
+  have access to a communication channel that is encrypted and
+  authenticated between individual pairs of users and that also allows
+  uncensorable authenticated broadcast from each user to all other
+  users.  The _Chill distributed key generation_ (ChillDKG) protocol
```
"ChillDKG" is never expanded to "Chill distributed key generation" in either the ML post or the BIP.
```suggestion
  users.  The ChillDKG protocol
```

harding replied:
> Title: ChillDKG: Distributed Key Generation for FROST

https://github.com/BlockstreamResearch/bip-frost-dkg

jonasnick replied:
I had never thought about expanding ChillDKG, seems more like a name to me.

harding replied:
I'm very confused, but I'll remove the expansion.

vostrnad commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,187 @@
+  communication.
+
+  Encryption and authentication between participants starts with an
+  [elliptic curve diffie-hellman][ecdh] (ECDH) exchange.  Proof of
```
I know how we feel about uppercase names around here, but this still seems a bit egregious. (It's also inconsistent as all previous newsletters had "Diffie–Hellman" in uppercase.)
```suggestion
  [Elliptic-curve Diffie--Hellman][ecdh] (ECDH) exchange.  Proof of
```

harding replied:
My decision years ago to go lowercase on eponyms in protocol names was a weird one, for sure.  However, it seems increasingly inconsistent to me to do it only in some cases rather than all cases, so this is an attempt to be more consistent going forward.

How about I leave this as is for this newsletter, but if you feel strongly about it, you open an issue and we can discuss changing the general policy (plus optionally revising already published content).  If you do that, please @ me on the issue so I see it.

Gustavojfe commented:
all merge summaries have been added

LarryRuane commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,226 @@
+- [Core Lightning #7455][] enhances `connectd`'s [onion message][topic onion
+  messages] handling by implementing forwarding via both `short_channel_id`
+  (SCID) and `node_id`. Onion messages are now always enabled, and incoming
+  messages are rate limited to 4 per second. 
```
```suggestion
  messages are rate limited to 4 per second.
```

LarryRuane commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,226 @@
+  features to its peers, but has `route_blinding` disabled by default because it
+  will not forward [blinded payments][topic rv routing] that do not use
+  [trampoline routing][topic trampoline payments]. See Newsletter [#309][news309
+  quiescence] for an introduction to channel quiescence. 
```
```suggestion
  quiescence] for an introduction to channel quiescence.
```

LarryRuane commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,226 @@
+  bonds][news161 fidelity] in JoinMarket-style coinjoin transactions. Fidelity
+  bonds improve the sybil resistance of the protocol by creating a reputation
+  system where makers prove their intentional sacrifice of the time value of
+  money by timelocking bitcoin. 
```
```suggestion
  money by timelocking bitcoin.
```

LarryRuane commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,226 @@
+  messages][topic onion messages]. Nodes now ignore this field outside the
+  current payment to prevent fingerprinting of HTLC senders. The change aims to
+  discourage payment delays due to outdated channel parameters while still
+  allowing nodes with stale gossip data to benefit from updates when needed. 
```
```suggestion
  allowing nodes with stale gossip data to benefit from updates when needed.
```

LarryRuane reviewed (commented):
These are the trailing spaces that are preventing the deploy preview from succeeding.

jirijakes commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,226 @@
+
+  Encryption and authentication between participants starts with an
+  [elliptic curve diffie-hellman][ecdh] (ECDH) exchange.  Proof of
+  non-censorship is obtained by each participant using their base key to
+  sign a transcript of the session from its start to its end, where the
```
This wording sounds like each participant obtains the proof using his key however I believe it's the other way: each participant _sends_ the transcript signed by his key and all the others receive it and verify the signature.

Not sure what would be better wording, at least this could work:

```suggestion
  non-censorship is sent by each participant using their base key to
  sign a transcript of the session from its start to its end, where the
```

jonasnick commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,226 @@
+  participant's signed session transcript.
+
+  The goal was to design a fully generalized protocol that will be
+  usable in all cases where people want to generate keys for
+  Bitcoin-style scriptless threshold signatures.  Additionally, the
```
This is somewhat misleading. ChillDKG works for FROST, but it could be insecure when used in other signing schemes. See section ["Use ChillDKG only for FROST"](https://github.com/BlockstreamResearch/bip-frost-dkg?tab=readme-ov-file#use-chilldkg-only-for-frost). In short, this is because ChillDKG participants can bias the final key (which is fine for FROST).

jonasnick reviewed (commented):
ChillDKG section looks good except for the claim that ChillDKG can generate keys in any "scriptless" threshold signatures. Thanks!

harding commented:
Added lede, removed releases/RCs section (none this week), and added topic entries.  Made edits for all feedback (except where indicated by a comment reply), thanks @LarryRuane @bitschmidty @vostrnad @sipa @jirijakes @jonasnick !  Reviewed contributions by @bitschmidty and @Gustavojfe, leaving a few edits for the later (thanks!).

Thanks everyone!

bitschmidty commented on _topics/en/cluster-mempool.md:
```diff
@@ -21,6 +21,9 @@ primary_sources:
     - title: "Bitcoin Core #27677: new mempool design"
       link: https://github.com/bitcoin/bitcoin/issues/27677
 
+    - title: Introduce to cluster linearization
```
```suggestion
    - title: Introduction to cluster linearization
```

bitschmidty commented on _topics/en/timelocks.md:
```diff
@@ -64,6 +64,9 @@ optech_mentions:
   - title: "Soft fork proposal for fee-dependent timelocks"
     url: /en/newsletters/2024/01/03/#fee-dependent-timelocks
 
+  - title: "BIP26 added for timelocked fidelity bonds"
```
```suggestion
  - title: "BIP46 added for timelocked fidelity bonds"
```

bitschmidty commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,229 @@
+- [BDK #1489][] updates `bdk_electrum` to use merkle proofs for simplified
+  payment verification (SPV). It fetches merkle proofs and block headers alongside
+  transactions, validates that transactions are in confirmed blocks before
+  inserting anchors, and removes re-org handling from `full_scan`. The PR also
```
```suggestion
  inserting anchors, and removes reorg handling from `full_scan`. The PR also
```

bitschmidty commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,229 @@
+
+- [BIPs #1599][] adds [BIP46][] for a derivation scheme for HD wallets that
+  create [timelocked][topic timelocks] addresses used for [fidelity
+  bonds][news161 fidelity] as used for JoinMarket-style coinjoin market matching. Fidelity
```
```suggestion
  bonds][news161 fidelity] as used for JoinMarket-style [coinjoin][topic coinjoin] market matching. Fidelity
```

bitschmidty commented on _posts/en/newsletters/2024-07-19-newsletter.md:
```diff
@@ -0,0 +1,229 @@
+  underpay the encoded onion value. For example, Alice provides a lightning
+  invoice to Bob but she has no payment channels, so when Bob pays, Carol
+  (Alice’s LSP) creates a channel on the fly. To allow Carol to take a fee from
+  Alice to cover the cost of the initial on-chain fee that creates a [JIT channel][topic jit channels],
```
```suggestion
  Alice to cover the cost of the initial onchain fee that creates a [JIT channel][topic jit channels],
```
